pmmp/PocketMine-MP
# Caching creative inventory entries

## Introduction

I could notice during a debugging that the entries of the packet that is sent to the client to synchronize the creative menu, was recalculated each time.  

This is quite surprising because it is not linked to the player and therefore can only change if you add/remove items from the creative menu.

This difference is much bigger with an important number of items more than pocketmine base. 950 base items against 1700 for my case.

Implementing this caching system allows to reduce the sending of the creative inventory to just what is necessary: the encoding / network compression.

## Changes
### API changes

* Added ``CreativeInventoryCache``

## Tests

* Launch server
* Active timings
* Connect player
* I did my test by alternating a change of game mode from survival to creative, creative survival, 4 times (4 commands)
* Recover your timings

You should not see a significant difference with the basic pocketmine data.
